pallets/qpow/src/lib.rs

```diff
@@ -141,55 +124,49 @@ pub mod pallet {
 		ProofSubmitted {
 			nonce: NonceType,
 			difficulty: U512,
-			distance_achieved: U512,
+			hash_achieved: U512,
 		},
-		DistanceThresholdAdjusted {
-			old_distance_threshold: DistanceThreshold,
-			new_distance_threshold: DistanceThreshold,
+		DifficultyAdjusted {
+			old_difficulty: Difficulty,
+			new_difficulty: Difficulty,
 			observed_block_time: BlockDuration,
 		},
 	}
 
-	#[pallet::error]
-	pub enum Error<T> {
-		InvalidSolution,
-		ArithmeticOverflow,
-	}
-
-	pub fn get_initial_distance_threshold<T: Config>() -> DistanceThreshold {
-		DistanceThreshold::one().shl(T::InitialDistanceThresholdExponent::get())
+	pub fn get_initial_difficulty<T: Config>() -> Difficulty {
+		T::InitialDifficulty::get()
 	}
 
 	#[pallet::hooks]
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		// TODO: update this
```

Comment:
@dastansam can you review these hooks and see if they still make sense? I suspect on_idle is redundant now